core/src/main/java/org/geysermc/geyser/translator/inventory/OldSmithingTableTranslator.java

```diff
@@ -0,0 +1,141 @@
+        super(3, "minecraft:smithing_table", ContainerType.SMITHING_TABLE, UIInventoryUpdater.INSTANCE);
+    }
+
+    @Override
+    public int bedrockSlotToJava(ItemStackRequestSlotData slotInfoData) {
+        return switch (slotInfoData.getContainer()) {
+            case SMITHING_TABLE_INPUT -> 0;
+            case SMITHING_TABLE_MATERIAL -> 1;
+            case SMITHING_TABLE_RESULT, CREATED_OUTPUT -> 2;
+            default -> super.bedrockSlotToJava(slotInfoData);
+        };
+    }
+
+    @Override
+    public BedrockContainerSlot javaSlotToBedrockContainer(int slot) {
+        return switch (slot) {
+            case 0 -> new BedrockContainerSlot(ContainerSlotType.SMITHING_TABLE_INPUT, 51);
+            case 1 -> new BedrockContainerSlot(ContainerSlotType.SMITHING_TABLE_MATERIAL, 52);
+            case 2 -> new BedrockContainerSlot(ContainerSlotType.SMITHING_TABLE_RESULT, 50);
+            default -> super.javaSlotToBedrockContainer(slot);
+        };
+    }
+
+    @Override
+    public int javaSlotToBedrock(int slot) {
+        return switch (slot) {
+            case 0 -> 51;
+            case 1 -> 52;
+            case 2 -> 50;
+            default -> super.javaSlotToBedrock(slot);
+        };
+    }
+
+    @Override
+    public boolean shouldHandleRequestFirst(ItemStackRequestAction action, Inventory inventory) {
+        return true;
+    }
+
+    @Override
+    protected ItemStackResponse translateSpecialRequest(GeyserSession session, Inventory inventory, ItemStackRequest request) {
+        for (var action: request.getActions()) {
+            switch (action.getType()) {
+                case DROP -> {
+                   if (((DropAction) action).getSource().getContainer().equals(ContainerSlotType.SMITHING_TABLE_TEMPLATE)) {
+                       return rejectRequest(request, false);
+                   }
+                }
+                case TAKE -> {
+                    if (((TakeAction) action).getSource().getContainer().equals(ContainerSlotType.SMITHING_TABLE_TEMPLATE)
+                    || ((TakeAction) action).getDestination().getContainer().equals(ContainerSlotType.SMITHING_TABLE_TEMPLATE)) {
+                        return rejectRequest(request, false);
+                    }
+                }
+                case SWAP -> {
+                    if (((SwapAction) action).getSource().getContainer().equals(ContainerSlotType.SMITHING_TABLE_TEMPLATE)
+                    || ((SwapAction) action).getDestination().getContainer().equals(ContainerSlotType.SMITHING_TABLE_TEMPLATE)) {
+                        return rejectRequest(request, false);
+                    }
+                }
+                case PLACE -> {
+                    if (((PlaceAction) action).getSource().getContainer().equals(ContainerSlotType.SMITHING_TABLE_TEMPLATE) ||
+                            ((PlaceAction) action).getDestination().getContainer().equals(ContainerSlotType.SMITHING_TABLE_TEMPLATE)) {
```

Comment:
I would introduce a `isInvalidAction(...)` method to this class that checks if the destination or source is the template slot